### ISSUE TITLE
Pytorch: calling `detach` on a tensor throws a wrong type error

### ISSUE THREAD

**Issue by levnikmyskin** [zubanls/zuban#127]
As per title, minimal example to reproduce:
```python
import torch as th

th.tensor([1,2,3,4]).detach()
# zubanls: Attribute function "detach" with type "Callable[[], Tensor]" does not accept self argument [misc]
```

**Comment by davidhalter** [comment]
Could you give me the reason why you think this is not correct? Or at least give me the link to the detach function? I know that this is probably an issue, but if you don't do that I have to find the detach function in the tensorflow code base, which is extremely big and I have no knowledge about.

Most people that report issues here try to isolate it from dependencies as well, which is sometimes hard and I don't expect it, but it's just a lot of work to work through issues like this one.

**Comment by levnikmyskin** [comment]
hi! Well sure, I'll try to help as I can: this is not correct simply because you do call `detach` on `tensor` objects, that's how the library works.
I found the documentation here: https://github.com/pytorch/pytorch/blob/3f1824742cac2ffb9a3afd90953c492c6c7f2f50/torch/_tensor.py#L804
but not entirely sure where zuban would read the type from. I can look into it a bit more if necessary

**Comment by davidhalter** [comment]
Yeah that's the hard part :) You probably have to look around in the `.pyi` files. But some of those pyi files are generated. You therefore probably should look locally and not in the Github repo. I'm pretty sure the part you linked is not what's used for type checking. You can probably use goto on `detach` to find the file for the type. But you need to use `goto type definition` and not the normal goto.
